lib/dispatcher/agent.js

```diff
@@ -110,6 +110,17 @@ class Agent extends DispatcherBase {
 
     await Promise.all(destroyPromises)
   }
+
+  get stats () {
```

Comment:
Getter to loop over Clients which can be Pool or Client themselves to assemble and expose stats.

Data structure can be changed. Right now it's `[string, [PoolStats, ClientStats]]`. It could also be `Record<string, PoolStats | ClientStats>`.

Reply:
The current shape seems like a good first iteration, especially for Agent as the number of origins called can vary; caller can reconstruct it into a different data structure if needed

Reply:
Thanks. Same thinking on my end.